backend/tarsy/agents/iteration_controllers/react_tools_controller.py

```
@@ -0,0 +1,171 @@
+Please provide a final summary of the data collected, even if the collection isn't complete."""
+        
+        try:
+            messages = [
+                LLMMessage(
+                    role="system", 
+                    content="Provide a summary of data collected based on the investigation."
+                ),
+                LLMMessage(role="user", content=final_prompt)
+            ]
+            
+            fallback_response = await self.llm_client.generate_response(messages, context.session_id, context.agent.get_current_stage_execution_id())
```

Ruff (0.12.2)

Line too long (149 > 88)

(E501)